src/app/services/login-route.guard.ts

```diff
@@ -0,0 +1,28 @@
+import { Router, CanActivate } from "@angular/router";
+import { Injectable } from "@angular/core";
+import { LoginService } from './login.service';
+import { Location } from "@angular/common";
+
+@Injectable()
+export class LoginRouteGuard implements CanActivate {
+
+    constructor(
+        private router: Router,
+        private location: Location,
+        private loginService: LoginService,
+    ) { }
+
+    canActivate() {
+        if ( this.loginService.isLoggedIn() ) {
+            return true;
+        }
+
+        // this.location.back(); //TODO: How to retain state?
```

Comment:
Could you elaborate on this question?

Reply:
I can't quite recall, but when I had the back() called, the Search History and Shopping cart data would be lost. How do I persist the store even during router navigations, user browser actions, etc? Is it storing redux state data via localStorage?